plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java

```diff
@@ -0,0 +1,198 @@
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testSelect()
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+
+        getDistributedQueryRunner().execute(TEST_SESSION, "SELECT * FROM test_select_from");
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 14,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testSelectWithFilter()
+    {
+        assertUpdate("CREATE TABLE test_select_from_where AS SELECT 2 as age", 1);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+
+        assertQuery("SELECT * FROM test_select_from_where WHERE age = 2", "VALUES " + 2);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 14,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testJoin()
+    {
+        assertUpdate("CREATE TABLE test_join_t1 AS SELECT 2 as age, 'id1' AS id", 1);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+        assertUpdate("CREATE TABLE test_join_t2 AS SELECT 'name1' as name, 'id1' AS id", 1);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 2,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 4,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+
+        MaterializedResult materializedRows = computeActual("SELECT name, age FROM test_join_t1 JOIN test_join_t2 ON test_join_t2.id = test_join_t1.id");
+        assertEqualsIgnoreOrder(materializedRows, resultBuilder(getSession(), VARCHAR, IntegerType.INTEGER)
+                .row("name1", 2)
+                .build()
+                .getMaterializedRows());
```

Comment:
I am on the fence whether we need to check result correctness here.

- it's not material from test perspective, so we could skip it
- we can still do it to ensure there is nothing bogus about our test queries (eg by means of a typo)
  - if we choose this path, i wonder how we can explain the reasoning in the code. a comment somewhere?

Reply:
i was wondering too if this is necessary, I will skip it